File: java/src/com/mogreet/mercury/user/Info.java
/**
* Copyright 2011 Nicolas Payot
*
* This file is part of Mercury.
*
* Mercury is free software: you can redistribute it and/or modify
* it under the terms of the GNU General Public License as published by
* the Free Software Foundation, either version 3 of the License, or
* (at your option) any later version.
*
* Mercury is distributed in the hope that it will be useful,
* but WITHOUT ANY WARRANTY; without even the implied warranty of
* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
* GNU General Public License for more details.
*
* You should have received a copy of the GNU General Public License
* along with Mercury.  If not, see <http://www.gnu.org/licenses/>.
*/

package com.mogreet.mercury.user;

import com.mogreet.mercury.*;
import org.w3c.dom.*;
import javax.xml.xpath.*;
import java.lang.Math.*;

/**
* The Info object contains the response from an {@link Mercury#info} request to the Moms API.
*/
public class Info extends Response {
		
	private String number;
	private int carrierId;
	private String carrier;
	private int handsetId;
	private String handset;
	
	/**
	* Creates a new Info response object.
	* @param xmlDoc XML DOM document that is returned by all the requests.
	*/
	public Info(Document xmlDoc) {
		super(xmlDoc);
		if (!super.responseIsValid())
			throw new RuntimeException(super.getMessage());
		try {
			XPath xpath = XPathFactory.newInstance().newXPath();
			XPathExpression expr;

			expr = xpath.compile("/response/number/text()");
			this.number = (String)(expr.evaluate(xmlDoc, XPathConstants.STRING));
			
			expr = xpath.compile("/response/carrier/@id");
			this.carrierId = ((Double)(expr.evaluate(xmlDoc, XPathConstants.NUMBER))).intValue();
			
			expr = xpath.compile("/response/carrier/text()");
			this.carrier = (String)(expr.evaluate(xmlDoc, XPathConstants.STRING));
			
			expr = xpath.compile("/response/handset/@id");
			this.handsetId = ((Double)(expr.evaluate(xmlDoc, XPathConstants.NUMBER))).intValue();
			
			expr = xpath.compile("/response/handset/text()");
			this.handset = (String)(expr.evaluate(xmlDoc, XPathConstants.STRING));

		} catch (XPathExpressionException e) {
			throw new RuntimeException("\nAn error occured while parsing the XML data for the INFO call: " + e.getMessage());
		}
	}
	
	/**
	* @return The requested number.
	*/
	public String getNumber() {
		return this.number;
	}
	
	/**
	* @return The carrier ID for the requested number.
	*/
	public int getCarrierId() {
		return this.carrierId;
	}
	
	/**
	* @return The carrier name for the requested number.
	*/
	public String getCarrier() {
		return this.carrier;
	}
	
	/**
	* @return The handset ID for the requested number.
	*/
	public int getHandsetId() {
		return this.handsetId;
	}
	
	/**
	* @return The handset name for the requested name.
	*/
	public String getHandset() {
		return this.handset;
	}
}

File: dotNet/MogreetSDK/MogreetSDK/user/Info.cs
using System;
using System.Xml;
using System.Security;
using System.Collections.Generic;

namespace MogreetSDK.user
{
    /// <summary>
    /// The Info object contains the response from an <see cref="MogreetSDK.Mercury.info"/> request to the Moms API.
    /// </summary>
    /// <exception cref='Exception'>
    /// Represents errors that occur during application execution.
    /// </exception>
    public class Info : Response
    {
        private string number;
        private int carrierId;
        private string carrier;
        private int handsetId;
        private string handset;

        /// <summary>
        /// Initializes a new instance of the <see cref="SDK.Info"/> class.
        /// </summary>
        /// <param name='xmlDoc'>
        /// Xml document that is returned by all the requests.
        /// </param>
        /// <exception cref='Exception'>
        /// Represents errors that occur during application execution.
        /// </exception>
        public Info(XmlDocument xmlDoc)
            : base(xmlDoc)
        {
            if (!(base.responseIsValid()))
                throw new Exception(base.getMessage());
            try
            {
                // Creates XML Reader
                XmlNode numr = mXmlDoc.SelectSingleNode("//response/number/text()");
                this.number = numr.Value;
                Console.WriteLine("number: " + this.number);

                XmlNode caridr = mXmlDoc.SelectSingleNode("//response/carrier/@id");
                this.carrierId = Int32.Parse(caridr.InnerText);
                Console.WriteLine("carrierId: " + this.carrierId);

                XmlNode carr = mXmlDoc.SelectSingleNode("//response/carrier/text()");
                this.carrier = carr.Value;
                Console.WriteLine("carrier: " + this.carrier);

                XmlNode hsidr = mXmlDoc.SelectSingleNode("//response/handset/@id");
                this.handsetId = Int32.Parse(hsidr.InnerText);
                Console.WriteLine("handsetId: " + this.handsetId);

                XmlNode hsr = mXmlDoc.SelectSingleNode("//response/handset/text()");
                this.handset = hsr.Value;
                Console.WriteLine("handset: " + this.handset);

            }
            catch (XmlException e)
            {
                throw new Exception("\nAn error occured while parsing the XML data for the INFO call: " + e.Message);
            }
        }

        /// <summary>
        /// Gets the number.
        /// </summary>
        /// <returns>
        /// The requested number.
        /// </returns>
        public string getNumber()
        {
            return this.number;
        }

        /// <summary>
        /// Gets the carrier identifier.
        /// </summary>
        /// <returns>
        /// The carrier ID for the requested number.
        /// </returns>
        public int getCarrierId()
        {
            return this.carrierId;
        }

        /// <summary>
        /// Gets the carrier.
        /// </summary>
        /// <returns>
        /// The carrier name for the requested number.
        /// </returns>
        public string getCarrier()
        {
            return this.carrier;
        }

        /// <summary>
        /// Gets the handset identifier.
        /// </summary>
        /// <returns>
        /// The handset ID for the requested number.
        /// </returns>
        public int getHandsetId()
        {
            return this.handsetId;
        }

        /// <summary>
        /// Gets the handset.
        /// </summary>
        /// <returns>
        /// The handset name for the requested name.
        /// </returns>
        public string getHandset()
        {
            return this.handset;
        }
    }
}
